### ISSUE TITLE
[Feature request] - (UI Visual Effects Update Proposal )

### ISSUE THREAD

**Issue by PatrickAlex2019** [tannerhelland/PhotoDemon#397]
Hope the overall interface can adapt to the rounded mica effect of windows11

**Comment by tannerhelland** [comment]
Hello @PatrickAlex2019 .  Thank you for submitting this issue.

PhotoDemon needs to provide a uniform interface on everything from Windows XP through the latest Windows 11 builds.  To do this, it tries to find a "middle-ground" between the aesthetics of the various Windows releases, while still supporting critical features like light and dark themes.  

Most UI changes to PhotoDemon are tackled when there is a clear benefit to usability.  Do you feel the rounded corners in Windows 11 make apps easier to use?  (This is not a sarcastic question - I am genuinely asking.  I have not noticed a meaningful difference in my own time with Windows 11, but I am open to other's feedback.)

A UI overhaul to use rounded corners instead of square ones would be a large project, and I need to make sure it would truly benefit users before spending my time there.  Thank you in advance for any follow-up information you can provide.

**Comment by p-yukusai** [comment]
I believe it is more of it being just more aesthetically pleasing for a lot of people, myself included, one thing however that I believe would improve usability relating to a transition to the Mica UI or something similar to it is the addition of a darker title bar for the application, as well as more coherence of the application's theming overall, since utilizing it with the built-in dark theme can be a bit jarring, at least in my opinion, as whenever I utilize wish to utilize an effect my eyes jump from a darker palette to pure white and i find that kind of contrast to make it harder to discern certain shades of colors for me.

**Comment by Kroc** [comment]
Application theming is very difficult to achieve with VB6 and requires manually drawing the UI. This adds a lot of code that takes a long time to develop and an immense amount of additional testing. If Tanner does pursue this, progress will be slow for the time being. I would prefer a dark menu too, but having worked with custom menus in VB6 in the past too, I'm aware of what an undertaking this is.

There may yet be some small additional automatic theming on Windows 11 that can be opted into through the application manifest but this is only assumption on my part.

**Comment by p-yukusai** [comment]
Yeah Visual Basic isn't great at drawing UI components, that's something more suited to something like C#.
The fact that PD works and looks as well as it does is amazing as is, and something like what I proposed would probably mean making the window borderless and then coding the title bar by scratch, a very annoying process as I have learned with a higher level language, so I can't even imagine how hard it would be on VBA.

**Comment by tannerhelland** [comment]
@p-yukusai and @Kroc thank you for the follow-up discussion!

Themed menus and title-bars are on my TODO list and I completely agree: PhotoDemon would benefit greatly from them.  I have already merged code to activate a dark title-bar but it looks atrocious with PhotoDemon's current bright menus, so I have disabled that feature until PhotoDemon's menus can be properly dark-themed.

The biggest challenge in theming PhotoDemon's menus is that I can't tackle it piecemeal.  I have to completely remove the current menu system and build a new one from scratch.  This is time-consuming and challenging to accurately debug across all supported versions of Windows (e.g. XP has totally different constraints than Win 11 here).

I will try to update this thread when themed menus and title-bars near completion.  Thank you to everyone for their patience 🙏
